factory.js

```diff
@@ -0,0 +1,10 @@
+training_set = null;
```

Comment:
These two vars are a bit weird.
Is there a reason these are declared implicitly global (without var, let or const)?

Is there a way to avoid global vars? Or is this required for the whole `vm2`stuff?

When it is required:
I think it should work to make these explicitly global by using `global.training_set = null;`, right?

Reply:
Oh yeah sorry for that. I was trying something else originally. Will change this

Reply:
fixed